src/fsharp/PostInferenceChecks.fs

```diff
@@ -1776,6 +1776,21 @@ let CheckModuleBinding cenv env (TBind(v,e,_) as bind) =
                            MethInfosEquivByNameAndSig EraseAll true g cenv.amap v.Range minfo1 minfo2 then 
                             errorR(Duplicate(kind,v.DisplayName,v.Range)))
 
+            if v.IsExtensionMethod then
+                let minfo = FSMeth(g, generalizedTyconRef tcref, mkLocalValRef v, Some 0UL)
+                minfo.TryHeadObjArgsByrefType(cenv.amap, v.Range, minfo.FormalMethodInst)
+                |> Option.iter (fun ty ->
+                    let hasStructOrStructConstraint =
```

Comment:
Code cleanup: Predicates defining new logic over types should always be split out into new functions in TastOps.fs/fsi.  For example, make this `isStructOrStructConstraintTy` and we should at some point compare all uses of `isStructTy` to check if there the two can be merged.  I've almost never seen F# code use `'T : struct` except when inferred from `Nullable<_>` so it may be an under-tested part of the logic (though to be honest not desperately urgent to fix)

Reply:
Makes sense to me.